File: projects/order-creation-queries/app/src/main/java/com/robinfood/app/usecases/getsuggestedportions/IGetSuggestedPortionsUseCase.java
package com.robinfood.app.usecases.getsuggestedportions;

import com.robinfood.core.dtos.menu.MenuSuggestedPortionResponseDTO;
import com.robinfood.core.enums.Result;

import java.util.List;

/**
 * Use case to get the suggested portions for given products
 */
public interface IGetSuggestedPortionsUseCase {

    /**
     * Retrieve the suggested portions
     *
     * @param portionIds list with the portions identifier
     * @param token the authentication token to be used
     * @return the change portions
     */
    Result<List<MenuSuggestedPortionResponseDTO>> invoke(
            List<Long> portionIds,
            String token
    );
}

File: projects/posv2/app/src/main/kotlin/com/robinfood/app/usecases/menuproductdetail/IGetSuggestedPortionsUseCase.kt
package com.robinfood.app.usecases.menuproductdetail

import com.robinfood.core.dtos.menu.response.MenuSuggestedPortionResponseDTO
import com.robinfood.core.enums.Result

/**
 *  Use case that retrieves the suggested portions
 */
interface IGetSuggestedPortionsUseCase {

    /**
     * Retrieve the suggested portions
     * [token] the authentication token to be used
     * [portionIds] list with the portions identifier
     * @return the change portions
     */
    suspend fun invoke(
            token: String,
            portionIds: List<Long>
    ): Result<List<MenuSuggestedPortionResponseDTO>>
}